# GUIA 1 - ACTIVIDAD 1. FUNCIONES Y TIPOS DE VARIABLES

# Uso de pandas para la importación de archivo.

In [5]:
import pandas as pd #Importamos la biblioteca de pandas para el uso y analisis de los datos de un dataframe.

Importando datos y asignamos el dataframe a la variable censu

In [7]:
censu=pd.read_csv("C:/Users/Adm1n/Maestria_EAN/census_income.csv",engine='python') #Incluimos el parametro engine para determinar que estamos usando python

Tipos de variables que contienen el archivo

In [9]:
censu.info() #Aplicamos la funcion info() donde nos da la informacion del dataframe.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Unnamed: 0      48842 non-null  int64 
 1   age             48842 non-null  int64 
 2   workclass       47879 non-null  object
 3   fnlwgt          48842 non-null  int64 
 4   education       48842 non-null  object
 5   education-num   48842 non-null  int64 
 6   marital-status  48842 non-null  object
 7   occupation      47876 non-null  object
 8   relationship    48842 non-null  object
 9   race            48842 non-null  object
 10  sex             48842 non-null  object
 11  capital-gain    48842 non-null  int64 
 12  capital-loss    48842 non-null  int64 
 13  hours-per-week  48842 non-null  int64 
 14  native-country  48568 non-null  object
 15  income          48842 non-null  object
dtypes: int64(7), object(9)
memory usage: 6.0+ MB


Como nos informa la función .info() el archivo censu.csv contiene 4.8842 filas y 16 columnas en total de las cuales tenemos 7 variables de tipo entero(int) y 9 variables de tipo cadena (str). 

#  Desplegamos las primeras filas del archivo

In [11]:
censu.head(100) # Aplicamos la función head() para poder visualizar nuestro dataframe. 
             # Al dejar vacio las variables por defecto .head() nos muestra las primeras 5 filas.

Unnamed: 0  age         workclass  fnlwgt     education  education-num  \
0            0   39         State-gov   77516     Bachelors             13   
1            1   50  Self-emp-not-inc   83311     Bachelors             13   
2            2   38           Private  215646       HS-grad              9   
3            3   53           Private  234721          11th              7   
4            4   28           Private  338409     Bachelors             13   
..         ...  ...               ...     ...           ...            ...   
95          95   29         Local-gov  115585  Some-college             10   
96          96   48  Self-emp-not-inc  191277     Doctorate             16   
97          97   37           Private  202683  Some-college             10   
98          98   48           Private  171095    Assoc-acdm             12   
99          99   32       Federal-gov  249409       HS-grad              9   

        marital-status         occupation   relationship   race     sex  \
0        Never-married       Adm-clerical  Not-in-family  White    Male   
1   Married-civ-spouse    Exec-managerial        Husband  White    Male   
2             Divorced  Handlers-cleaners  Not-in-family  White    Male   
3   Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4   Married-civ-spouse     Prof-specialty           Wife  Black  Female   
..                 ...                ...            ...    ...     ...   
95       Never-married  Handlers-cleaners  Not-in-family  White    Male   
96  Married-civ-spouse     Prof-specialty        Husband  White    Male   
97  Married-civ-spouse              Sales        Husband  White    Male   
98            Divorced    Exec-managerial      Unmarried  White  Female   
99       Never-married      Other-service      Own-child  Black    Male   

    capital-gain  capital-loss  hours-per-week native-country income  
0           2174             0              40  United-States  <=50K  
1              0             0              13  United-States  <=50K  
2              0             0              40  United-States  <=50K  
3              0             0              40  United-States  <=50K  
4              0             0              40           Cuba  <=50K  
..           ...           ...             ...            ...    ...  
95             0             0              50  United-States  <=50K  
96             0          1902              60  United-States   >50K  
97             0             0              48  United-States   >50K  
98             0             0              40        England  <=50K  
99             0             0              40  United-States  <=50K  

[100 rows x 16 columns]

# Detección de Valores Nulos

Para saber cuantos valores perdidos tenemos en cada una de las columnas (variables). Debemos usar la función .isnull().sum() la razon por la cual usamos en conjunto la función .sum() es por que va a estar sumando todos los valores que cumplan con la condicion null en cada una de las columnas.

In [23]:
valores_null=censu.isnull().sum()
print(valores_null)

Unnamed: 0          0
age                 0
workclass         963
fnlwgt              0
education           0
education-num       0
marital-status      0
occupation        966
relationship        0
race                0
sex                 0
capital-gain        0
capital-loss        0
hours-per-week      0
native-country    274
income              0
dtype: int64


Como podemos observar en la variable 'workclass' tiene 963 con valores nulos (NaN). Tambien la variable 'occupation' con 966 valores nulos y la variable 'native-country' con 274 valores nulos. Para un total de 2.203 datos.

# Detección filas duplicadas

Aplicamos la función .duplicated() la cual nos dice mediante una confirmación con numeros booleanos si existe o no esta condición. De igual forma, adicionamos la función .sum() para determiar a lo largo de todas las filas la condición de filas duplicadas.

In [25]:
filas_duplica=censu.duplicated().sum()
print(filas_duplica)

0


Como podemos observar ninguna fila tiene información duplicada en el dataframe de censu.

## Valores unicos

Aplicamos la función .nunique() al dataframe censu y le asignamos la variable cant_unicos para posteriormente imprimir el resultado con la función print()

In [19]:
# Calcular cuantos valores unicos hay 
cant_unicos = censu.nunique()
print(cant_unicos)

Unnamed: 0        48842
age                  74
workclass             9
fnlwgt            28523
education            16
education-num        16
marital-status        7
occupation           15
relationship          6
race                  5
sex                   2
capital-gain        123
capital-loss         99
hours-per-week       96
native-country       42
income                4
dtype: int64


## Descripción estadistica del Dataframe

In [31]:
censu.describe() # Verificamos las operaciones estadisticas principales del dataframe censu

Unnamed: 0           age        fnlwgt  education-num  capital-gain  \
count  48842.000000  48842.000000  4.884200e+04   48842.000000  48842.000000   
mean   24420.500000     38.643585  1.896641e+05      10.078089   1079.067626   
std    14099.615261     13.710510  1.056040e+05       2.570973   7452.019058   
min        0.000000     17.000000  1.228500e+04       1.000000      0.000000   
25%    12210.250000     28.000000  1.175505e+05       9.000000      0.000000   
50%    24420.500000     37.000000  1.781445e+05      10.000000      0.000000   
75%    36630.750000     48.000000  2.376420e+05      12.000000      0.000000   
max    48841.000000     90.000000  1.490400e+06      16.000000  99999.000000   

       capital-loss  hours-per-week  
count  48842.000000    48842.000000  
mean      87.502314       40.422382  
std      403.004552       12.391444  
min        0.000000        1.000000  
25%        0.000000       40.000000  
50%        0.000000       40.000000  
75%        0.000000       45.000000  
max     4356.000000       99.000000

Podemos concluir que el estudio que se realizo en el rango de edades entre 17 90 años con un promedio de 38 años y una desviación estandar del 13.71 años. Las horas trabajadas por semana en promedio son de 40 horas con un minimo de 1 hora y un maximo de 99 horas.